web/server/codechecker_server/auth/cc_ldap.py

```diff
@@ -238,7 +244,8 @@ def __init__(self, ldap_config, who=None, cred=None):
         ldap.protocol_version = ldap.VERSION3
 
         # Check cert if available but do not fail if not.
-        ldap.set_option(ldap.OPT_X_TLS_REQUIRE_CERT, ldap.OPT_X_TLS_ALLOW)
+        if os.environ.get("LDAPTLS_REQCERT") == "never":
```

Comment:
I am thinking to move this option to the config file instead of environment variable similar to `deref` config option and others:
https://github.com/Ericsson/codechecker/blob/master/web/server/config/server_config.json#L31

The name can be for example `tls_require_cert`.
The handled values can be `never` and `allow`.
The default value can be `allow` to not change the behavior of the server. So who already use it, they will not need to change anything.

Also extend the documentation with the new config option: https://github.com/Ericsson/codechecker/blob/master/docs/web/authentication.md

Reply:
@csordasmarton according to the [python-ldap documentation](https://www.python-ldap.org/en/python-ldap-3.3.0/reference/ldap.html#ldap.OPT_X_TLS_REQUIRE_CERT) the value `allow` that I removed is only to be used internally by the slapd server. So I don't think I should re-introduce it. From our perspective, it seems to be equivalent to "never" since it allows insecure connections to go through.

I suggest the new option only handles one value: `never`. In that case no certificate verification happens, like today. In every other case, no option is passed to the ldap library using OPT_X_TLS_REQUIRE_CERT (which means secure behaviour).

This will break existing insecure LDAPS setups (will require admins to modify configuration to add `tls_require_cert="never"`) but it's probably better to do it sooner rather than later, since the ldap module doesn't seem widely used yet (it's missing core functionality for us at least).